test/e2e/vmagent_test.go

```diff
@@ -0,0 +1,297 @@
+	rc, ifIndex, stderr, err := data.RunCommandOnNode(nodeName, cmd)
+	require.NoError(t, err, "Failed to run command <%s> on VM %s, err %v", cmd, nodeName, err)
+	require.Equal(t, 0, rc, "Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, ifIndex, stderr)
+
+	vm.ifIndex = strings.TrimSpace(ifIndex)
+	cmd = fmt.Sprintf("powershell 'Get-NetAdapter -IfIndex %s | select name | ft -HideTableHeaders'", vm.ifIndex)
+	rc, ifName, stderr, err := data.RunCommandOnNode(nodeName, cmd)
+	require.NoError(t, err, "Failed to run command <%s> on VM %s, err %v", cmd, nodeName, err)
+	require.Equal(t, 0, rc, "Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, ifName, stderr)
+
+	vm.ifName = strings.TrimSpace(ifName)
+	cmd = fmt.Sprintf("powershell 'Get-NetIPAddress -AddressFamily IPv4 -ifIndex %s| select IPAddress| ft -HideTableHeaders'", vm.ifIndex)
+	rc, ifIP, stderr, err := data.RunCommandOnNode(nodeName, cmd)
+	require.NoError(t, err, "Failed to run command <%s> on VM %s, err %v", cmd, nodeName, err)
+	require.Equal(t, 0, rc, "Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, ifIP, stderr)
+
+	vm.ip = strings.TrimSpace(ifIP)
+	return vm
+
+}
+
+func startAntreaAgent(t *testing.T, data *TestData, vm vmInfo) {
+	t.Logf("Starting antrea-agent on VM: %s", vm.nodeName)
+	var cmd string
+	if vm.osType == "Windows" {
+		cmd = "nssm start antrea-agent"
+	} else {
+		cmd = "sudo systemctl start antrea-agent"
+	}
+	rc, stdout, stderr, err := data.RunCommandOnNode(vm.nodeName, cmd)
+	require.NoError(t, err, "Failed to run command <%s> on VM %s, err %v", cmd, vm.nodeName, err)
+	require.Equal(t, 0, rc, "Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, stdout, stderr)
+}
+
+func stopAntreaAgent(t *testing.T, data *TestData, vm vmInfo) {
+	t.Logf("Stopping antrea-agent on VM: %s", vm.nodeName)
+	var cmd string
+	if vm.osType == "Windows" {
+		cmd = "nssm stop antrea-agent"
+	} else {
+		cmd = "sudo systemctl stop antrea-agent"
+	}
+	rc, stdout, stderr, err := data.RunCommandOnNode(vm.nodeName, cmd)
+	require.NoError(t, err, "Failed to run command <%s> on VM %s, err %v", cmd, vm.nodeName, err)
+	require.Equal(t, 0, rc, "Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, stdout, stderr)
+}
+
+func verifyInterfaceIsInOVS(t *testing.T, data *TestData, vm vmInfo) (found bool, err error) {
+	var cmd string
+	if vm.osType == "Windows" {
+		cmd = fmt.Sprintf("ovs-vsctl --column=name list port '%s'", vm.ifName)
+	} else {
+		cmd = fmt.Sprintf("sudo ovs-vsctl --column=name list port %s", vm.ifName)
+	}
+	rc, stdout, stderr, err := data.RunCommandOnNode(vm.nodeName, cmd)
+	if err != nil {
+		return false, fmt.Errorf("failed to run command <%s> on VM %s, err %v", cmd, vm.nodeName, err)
+	}
+
+	if strings.Contains(stdout, "no row") {
+		return false, fmt.Errorf("failed to find OVS port %s on VM %s, err %v, rc %d, stdout %v, stderr %v", vm.ifName, vm.nodeName, err, rc, stdout, stderr)
```

Comment:
Do we expect the OVS port to be found immediately after the test starts? returnning error will terminate `PollImmediate` and return error, is it expected?

Reply:
Actually we dont want to return error and abort PollImmediate. 
But we do want to have special handling for this output. let me change return nil and just the log the output.